spec/services/update_auction_spec.rb

```diff
@@ -2,6 +2,43 @@
 
 describe UpdateAuction do
   describe '#perform' do
+    context 'when publishing an auction' do
+      it 'creates an AuctionState record' do
+        auction = create(:auction, :unpublished)
+
+        params = {
+          auction: {"published" => "published"}
+        }
+
+        expect do
+          UpdateAuction.new(
+            auction: auction,
+            params: params,
+            current_user: auction.user
+          ).perform
+        end.to change { AuctionState.count }.by(1)
+
+        auction.reload
+
+        # do we want to low-level test for the published state object, like this?
+        published_state = auction.states.find {|state| state.name == 'published'}
+        expect(published_state.state_value).to eq('published')
+
```

Comment:
^^^ that seems like the important part to test. You should probably also test what happens when an auction moves from 'archived' to 'published' in that case it will already have a record, so changing the count of records should not happen.

I forgot to mention that you can put a constraint in the database on that not happening by adding a uniq index on `['auction_id', 'name']`. Then if a unique index error is thrown you revert back to updating. Solves some obscure bugs that Rails generates via their expectation that there is always just one server running in one thread.